app/models/data_update_script.rb

```diff
@@ -0,0 +1,39 @@
+class DataUpdateScript < ApplicationRecord
+  DIRECTORY = Rails.root.join("lib/data_update_scripts").freeze
+  NAMESPACE = "DataUpdateScripts".freeze
+  STATUSES = %i[enqueued working succeeded failed].freeze
+
+  enum status: STATUSES
+  validates :file_name, uniqueness: true
+
+  def self.filenames
+    Pathname.
```

Comment:
Don't change it if you prefer the current version, but I find `Dir.glob` a bit easier in situations like this: 

```ruby
Dir.glob("*.rb", base: DIRECTORY).map { |f| Pathname.new(f).basename(".rb").to_s }
```